packages/core/parcel-bundler/src/builtins/prelude.js

```diff
@@ -103,5 +111,12 @@ parcelRequire = (function (modules, cache, entry, globalName) {
   }
 
   // Override the current require with this new one
+  parcelRequire = newRequire;
```

Comment:
This seems dangerous. Why are we overwriting the global variable here? Won't that already happen when the function returns?

Reply:
Because returning doesn't work if an error is thrown:

https://github.com/parcel-bundler/parcel/blob/75a71be0abe5082ca759680217ac128d2779b97d/packages/core/parcel-bundler/src/builtins/prelude.js#L114-L121

And this overwrites a global variable as well: 
https://github.com/parcel-bundler/parcel/blob/75a71be0abe5082ca759680217ac128d2779b97d/packages/core/parcel-bundler/src/builtins/prelude.js#L10